src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,127 @@
 package core.basesyntax;
 
-/**
- * Feel free to remove this class and create your own.
- */
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
 public class HelloWorldTest {
 
+    private RegistrationServiceImpl registrationService;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_validUser() {
+        User validUser = new User();
+        validUser.setAge(20);
+        validUser.setPassword("passwordValid");
+        validUser.setLogin("loginValid");
+
+        User result = registrationService.register(validUser);
+        assertNotNull(result);
+        assertEquals(validUser, storageDao.get("loginValid"));
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User firstUser = new User();
+        firstUser.setLogin("validLogin");
+        firstUser.setPassword("validPassword");
+        firstUser.setAge(20);
+        registrationService.register(firstUser);
+
+        User secondUserWithSameLogin = new User();
+        secondUserWithSameLogin.setLogin("validLogin");
+        secondUserWithSameLogin.setPassword("anotherPassword");
+        secondUserWithSameLogin.setAge(25);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(secondUserWithSameLogin);
+        });
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User userWithShortPassword = new User();
+        userWithShortPassword.setLogin("validLogin");
+        userWithShortPassword.setPassword("pwd");
+        userWithShortPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithShortPassword);
+        });
+    }
```

Comment:
You are testing the case where the password is too short, but you are not testing the edge case where the password is exactly the minimum length. It would be good to add a test for this case to ensure that the password length check is working correctly.